nengo/utils/graphs.py

```diff
@@ -98,6 +98,43 @@ def toposort(edges):
     return ordered
 
 
+def transitive_closure(edges, topo_sorted=None):
+    """Constructs the transitive_closure of a DAG.
+
+    The complexity is O(nodes + vertices).
+
+    Parameters
+    ----------
+    edges : dict
+        Dict of the form {a: {b, c}} where b and c depend on a. Must not
+        contain cycles.
+    topo_sorted : sequence, optional
+        The topological sorting of the vertices. If not passed in the algorithm
+        will do a topological sort.
+
+    Returns
+    -------
+    The transitive closure data structure as `edges`.
```

Comment:
Could you explain a bit more about what this is returning? From what I can tell, it's returning a dictionary where every key is an a node and the key is all the other nodes it can reach? Is that correct?